src/trades/TradesLayouts.js

```diff
@@ -9,10 +9,16 @@ export default class TradesLayouts extends Component {
         currency: PropTypes.string.isRequired,
         ticksForAllSymbols: PropTypes.object.isRequired,
         trades: PropTypes.array.isRequired,
+        tradesCount: PropTypes.number.isRequired,
         layoutN: PropTypes.number.isRequired,
         contracts: PropTypes.object.isRequired,
     };
 
+    componentWillMount() {
```

Comment:
Why? since layout is only relevant to TradeLayout, I think associate them make sense. We only want to generate those additional trade object when we want to render TradeLayout, right?

If you're talking about componentWillMount, I had the same concern, but after some research, I find it's fine, the only downside is that it will render one more time